### ISSUE TITLE
WebSQL cannot store more than 2GB data.

### ISSUE THREAD

**Issue by aakkas** [nwjs/nw.js#3094]
We are using WebSQL in our solution and we hit 2GB limit per origin. This limit is not per database. If we create 2 dbs we cannot exceed 2GB limit in total. 

I tried to find the problem and fix the issue however could not get NW to build. It seems build instructions are not complete or I missed a step somewhere.

If someone fix the issue that would be awesome.

Thanks
##

**Comment by Criten** [comment]
Is there a reason you aren't using a more robust storage solution? It seems like you are asking too much for WebSQL. Try just using the sqlite3 node bindings https://github.com/mapbox/node-sqlite3

:)

**Comment by aakkas** [comment]
Well WebSQL is implemented with SQLITE. It is true we would get more control when we use it from Node but the problem is we need to build those native libraries for each platform. I tried building sqlite with latest NW but compile failed. I believe earlier version of NW would work but whole process is pretty hard to follow. I wish NW would come with node sql by default.

**Comment by fritx** [comment]
@aakkas it's okay to build node-sqlite3 against NW.js, as I finally did it on both Windows and OSX.
See https://github.com/mapbox/node-sqlite3#building-for-node-webkit

However, node-sqlite3 has no data protection. I wonder whether websql, which built in Chrome, has such feature or not.

**Comment by Christywl** [comment]
Is this issue still reproduced on the latest nwjs build?

**Comment by Christywl** [comment]
close until there is more information.
